core/src/io/anuke/mindustry/net/Packets.java

```diff
@@ -140,51 +140,36 @@ public void read(ByteBuffer buffer) {
             timestamp = buffer.getLong();
         }
     }
-
-    public static class BlockLogSyncPacket implements Packet {
-		public IntMap<ArrayList<EditLog>> editlogs;
 	
+	public static class BlockLogRequestPacket implements Packet {
+		public int x;
+		public int y;
+		public Array<EditLog> editlogs;
+		
 		@Override
 		public void write(ByteBuffer buffer) {
-			int size = editlogs.size;
-			buffer.putInt(size);
-			for(IntMap.Entry<ArrayList<EditLog>> editlog :editlogs.entries()) {
-				
-				String key = String.valueOf(editlog.key);
-				buffer.put((byte) key.getBytes().length);
-				buffer.put(key.getBytes());
+			buffer.putInt(x);
```

Comment:
Not much of an issue, but you can write the position as a `short` instead of an `int`, since map sizes never get near `short_max`.